File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let pricemin = Infinity;
    let pricemax = 0;  

     for (let price of prices) {
        
        pricemin = Math.min(pricemin, price);

       
        pricemax = Math.max(pricemax, price - pricemin);
    }

    return pricemax;
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = Integer.MAX_VALUE;
        int max = 0;

        for(int value : prices){
            
            min = Math.min(min , value);
            max = Math.max(max, value - min);
        }
        return max;
    }
}